tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Minmod.cpp

```diff
@@ -137,30 +138,50 @@ Element<3> make_element() noexcept {
           {Direction<3>::upper_zeta(), make_neighbor_with_id<3>(6)}}};
 }
 
-std::unordered_map<Direction<1>,
-                   SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData>
+// Values of left_id, right_id based on make_element<1> above.
+std::unordered_map<std::pair<Direction<1>, ElementId<1>>,
+                   SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData,
+                   boost::hash<std::pair<Direction<1>, ElementId<1>>>>
 make_neighbor_packaged_data(
     const double left, const double right,
     const tnsr::I<double, 1>& local_size_for_neighbor_size) noexcept {
+  constexpr size_t left_id = 1;
+  constexpr size_t right_id = 2;
   return std::unordered_map<
-      Direction<1>, SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData>{
-      {Direction<1>::lower_xi(),
-       {Scalar<double>(left), local_size_for_neighbor_size}},
-      {Direction<1>::upper_xi(),
-       {Scalar<double>(right), local_size_for_neighbor_size}}};
+      std::pair<Direction<1>, ElementId<1>>,
+      SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData,
+      boost::hash<std::pair<Direction<1>, ElementId<1>>>>{
+      std::make_pair(
+          std::make_pair(Direction<1>::lower_xi(), ElementId<1>(left_id)),
+          SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData{
+              Scalar<double>(left), local_size_for_neighbor_size}),
+      std::make_pair(
+          std::make_pair(Direction<1>::upper_xi(), ElementId<1>(right_id)),
+          SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData{
+              Scalar<double>(right), local_size_for_neighbor_size})};
 }
 
-std::unordered_map<Direction<1>,
-                   SlopeLimiters::Minmod<1, tmpl::list<scalar>>::PackagedData>
+// Values of left_id, right_id based on make_element<1> above.
+std::unordered_map<std::pair<Direction<1>, ElementId<1>>,
```

Comment:
Yes, but I thought we preferred not using auto return. Am I remembering this wrong? or perhaps there is more a complex guildeline?

Reply:
`auto` return is not great because it forces the function definition to be visible at all call sites (but function in hpp). In this case all call sites are in the same source file so it doesn't matter.

Reply:
Thanks for the explanation!